pkg/services/ngalert/sender/router.go

```diff
@@ -257,27 +256,27 @@ func (d *AlertsRouter) buildExternalURL(ds *datasources.DataSource) (string, err
 func (d *AlertsRouter) Send(key models.AlertRuleKey, alerts definitions.PostableAlerts) {
 	logger := d.logger.New("rule_uid", key.UID, "org", key.OrgID)
 	if len(alerts.PostableAlerts) == 0 {
-		logger.Debug("no alerts to notify about")
+		logger.Info("No alerts to notify about")
 		return
 	}
 	// Send alerts to local notifier if they need to be handled internally
 	// or if no external AMs have been discovered yet.
 	var localNotifierExist, externalNotifierExist bool
 	if d.sendAlertsTo[key.OrgID] == models.ExternalAlertmanagers && len(d.AlertmanagersFor(key.OrgID)) > 0 {
-		logger.Debug("no alerts to put in the notifier")
+		logger.Debug("All alerts for the given org should be routed to external notifiers only. skipping the internal notifier.", "org", key.OrgID)
 	} else {
-		logger.Debug("sending alerts to local notifier", "count", len(alerts.PostableAlerts), "alerts", alerts.PostableAlerts)
```

Comment:
I think logging alerts is useful here but in the debug mode. Can you check if it is logged somewhere downstream?

Reply:
I'm not a huge fan of logging the entire postable alert, this contains all expanded annotation/label text which is a lot of user data in the logs. I'd prefer to find a middle ground here, perhaps we can do what the Mimir alertmanager does, which is log the alert name + truncated hash of labels?

Reply:
alert name does not make sense because we log UID. We need some info to be able to correlate an alert  in Grafana and in Alertmanager. If hash can do that, then let's do it.